privmx-endpoint/src/main/cpp/modules/KvdbApi.cpp

```diff
@@ -615,55 +615,115 @@ Java_com_simplito_java_privmx_1endpoint_modules_kvdb_KvdbApi_deleteEntries(
+                }
 
-    ctx.callVoidEndpointApi([&ctx, &thiz]() {
-        getKvdbApi(ctx, thiz)->unsubscribeFromKvdbEvents();
-    });
+                auto subscription_ids_c = getKvdbApi(ctx, thiz)->subscribeFor(
+                        subscription_queries_c);
+
+                jobject javaArrayList = env->NewObject(arrayListCls, initMID);
+                for (auto &id_str : subscription_ids_c) {
+                    jstring java_id_str = ctx->NewStringUTF(id_str.c_str());
+                    env->CallBooleanMethod(javaArrayList, addToListMID, java_id_str);
+                }
+
+                return javaArrayList;
+            }
+    );
+
+    if (ctx->ExceptionCheck()) {
+        return nullptr;
+    }
+    return result;
 }
-extern "C" JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_kvdb_KvdbApi_subscribeForEntryEvents(
+
+extern "C"
+JNIEXPORT void JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_kvdb_KvdbApi_unsubscribeFrom(
         JNIEnv *env,
         jobject thiz,
-        jstring kvdb_id
+        jobject subscription_ids
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(kvdb_id, "Kvdb ID")) {
+    if (ctx.nullCheck(subscription_ids, "Subscription IDs")) {
         return;
     }
 
-    ctx.callVoidEndpointApi([&ctx, &thiz, &kvdb_id]() {
-        getKvdbApi(ctx, thiz)->subscribeForEntryEvents(ctx.jString2string(kvdb_id));
+    ctx.callVoidEndpointApi([&ctx, &env, &thiz, &subscription_ids]() {
+        auto subscription_ids_arr = ctx.jObject2jArray(subscription_ids);
+        auto subscription_ids_c = std::vector<std::string>();
+
+        int length = ctx->GetArrayLength(subscription_ids_arr);
+        for (int i = 0; i < length; i++) {
+            jobject arrayElement = ctx->GetObjectArrayElement(subscription_ids_arr, i);
+            subscription_ids_c.push_back(ctx.jString2string((jstring) arrayElement));
+        }
+
+        getKvdbApi(ctx, thiz)->unsubscribeFrom(subscription_ids_c);
     });
 }
-extern "C" JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_kvdb_KvdbApi_unsubscribeFromEntryEvents(
+
+
+extern "C"
+JNIEXPORT jstring JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_kvdb_KvdbApi_buildSubscriptionQuery(
         JNIEnv *env,
         jobject thiz,
-        jstring kvdb_id
+        jlong eventType,
+        jlong selectorType,
+        jstring selectorId
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(kvdb_id, "Kvdb ID")) {
-        return;
+    if (ctx.nullCheck(selectorId, "SelectorID")) {
+        return nullptr;
     }
 
-    ctx.callVoidEndpointApi([&ctx, &thiz, &kvdb_id]() {
-        getKvdbApi(ctx, thiz)->unsubscribeFromEntryEvents(ctx.jString2string(kvdb_id));
-    });
+    jstring result = nullptr;
+    ctx.callResultEndpointApi<jstring>(
+            &result,
+            [&ctx, &env, &thiz, &eventType, &selectorType, &selectorId]() {
+                auto c_eventType = parseKvdbEventType(ctx, (long) eventType);
```

Comment:
you can easy cast long value to enum like this
```suggestion
                auto c_eventType = static_cast<kvdb::EventType>(eventType);
```